swift/Samples/Tutorial/Tutorial2.md

```diff
@@ -0,0 +1,597 @@
+
+```swift
+// MARK: Rendering
+
+extension WelcomeWorkflow {
+
+    typealias Rendering = WelcomeScreen
+
+    func render(state: WelcomeWorkflow.State, context: RenderContext<WelcomeWorkflow>) -> Rendering {
+
+        // Create a "sink" of type `Action`. A sink is what we use to send actions to the workflow.
+        let sink = context.makeSink(of: Action.self)
+
+        return WelcomeScreen(
+            name: state.name,
+            onNameChanged: { name in
+                sink.send(.nameChanged(name: name))
+            },
+            onLoginTapped: {
+                // Whenever the login button is tapped, emit the `.didLogin` action.
+                sink.send(.didLogin)
+            })
+    }
+}
+```
+
+Finally, map the output event from `WelcomeWorkflow` in `RootWorkflow` to the `login` action:
+
+```swift
+// MARK: Rendering
+
+extension RootWorkflow {
+
+    typealias Rendering = WelcomeScreen
+
+    func render(state: RootWorkflow.State, context: RenderContext<RootWorkflow>) -> Rendering {
+        // Render a child workflow of type `WelcomeWorkflow`. When `rendered(with:)` is called, the infrastructure will create
+        // a child workflow with state if one is not already running.
+        let welcomeScreen = WelcomeWorkflow()
+            .mapOutput({ output -> Action in
+                switch output {
+                    // When `WelcomeWorkflow` emits `didLogin`, turn it into our `login` action.
+                case .didLogin(name: let name):
+                    return .login(name: name)
+                }
+            })
+            .rendered(with: context)
+
+        return welcomeScreen
+    }
+}
+```
+
+### Showing a different workflow from state
+
+Now we are handling the `Output` of `WelcomeWorkflow`, and updating the state to show the `Todo` screen. However, we still need to update our render method to defer to a different workflow.
+
+We'll update the `render` method to show either the `WelcomeWorkflow` or `TodoListWorkflow` depending on the state of `RootWorkflow`
+
+Temporarily define the `Output` of `TodoListWorkflow` as `Never` (we can only go forward!):
+
+```swift
+// MARK: Input and Output
+
+struct TodoListWorkflow: Workflow {
+    typealias Output = Never
+}
+```
+
+And update the `render` method of the `RootWorkflow`:
+
+```swift
+// MARK: Rendering
+
+extension RootWorkflow {
+
+    typealias Rendering = AnyScreen
+
+    func render(state: RootWorkflow.State, context: RenderContext<RootWorkflow>) -> Rendering {
+        switch state {
+            // When the state is `.welcome`, defer to the WelcomeWorkflow
+        case .welcome:
+            // Render a child workflow of type `WelcomeWorkflow`. When `rendered(with:)` is called, the infrastructure will create
+            // a child workflow with state if one is not already running.
+            let welcomeScreen = WelcomeWorkflow()
+                .mapOutput({ output -> Action in
```

Comment:
IIRC we finished the naming summit unhappy with the `mapOutput` name but uncertain what to change it to? I guess I'll add a ticket reminding us to think about that some more.

Reply:
#583